### PR TITLE
Display file hashes into rabin via new mode.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: libr/core/cbin.c [hunk 1]
```diff
@@ -3459,6 +3459,24 @@ static int bin_header(RCore *r, int mode) {
...
 
+static int bin_hashes(RCore *r, int mode) {
+	ut64 lim = r_config_get_i (r->config, "cfg.hashlimit");
+	RIODesc *iod = r_io_desc_get (r->io, r->file->fd);
+	if (iod) {
+		r_bin_file_hash (r->bin, lim, iod->name);
+		if (IS_MODE_JSON (mode)) {
+			r_cons_printf ("{\"");
+		}
+		r_cons_printf (r_strbuf_get (r->bin->cur->o->info->hashes));
```
Comment (radare) [inline libr/core/cbin.c]: use "%s" or r_cons_strcat, but dont introduce a format string vulnerability in 2019

File: libr/core/cbin.c [hunk 2]
```diff
@@ -3459,6 +3459,24 @@ static int bin_header(RCore *r, int mode) {
 	return false;
 }
 
+static int bin_hashes(RCore *r, int mode) {
+	ut64 lim = r_config_get_i (r->config, "cfg.hashlimit");
```
Comment (radare) [inline libr/core/cbin.c]: maybe this should be under bin. isnt? didnt chked where it is used or how

File: libr/core/cbin.c [hunk 3]
```diff
@@ -3459,6 +3459,24 @@ static int bin_header(RCore *r, int mode) {
...
+	RIODesc *iod = r_io_desc_get (r->io, r->file->fd);
+	if (iod) {
+		r_bin_file_hash (r->bin, lim, iod->name);
+		if (IS_MODE_JSON (mode)) {
+			r_cons_printf ("{\"");
```
Comment (radare) [inline libr/core/cbin.c]: use the pj api

File: libr/include/r_bin.h [hunk 4]
```diff
@@ -62,6 +62,7 @@ R_LIB_VERSION_HEADER (r_bin);
 #define R_BIN_REQ_RESOURCES 0x8000000
 #define R_BIN_REQ_INITFINI 0x10000000
 #define R_BIN_REQ_SEGMENTS 0x20000000
+#define R_BIN_REQ_HASHES    0x40000000
```
Comment (radare) [inline libr/include/r_bin.h]: dont align that value

File: libr/include/r_core.h [hunk 5]
```diff
@@ -639,6 +639,7 @@ R_API void r_core_recover_vars(RCore *core, RAnalFunction *fcn, bool argonly);
 #define R_CORE_BIN_ACC_INITFINI 0x200000
 #define R_CORE_BIN_ACC_SEGMENTS 0x400000
 #define R_CORE_BIN_ACC_SOURCE 0x800000
+#define R_CORE_BIN_ACC_HASHES   0x10000000
```
Comment (radare) [inline libr/include/r_core.h]: same here

File: libr/core/cbin.c (hunk 1 above)
Comment (devnexen) [reply libr/core/cbin.c]: Ah right I took as a puts ...

File: libr/core/cbin.c [hunk 6]
```diff
@@ -3459,6 +3459,28 @@ static int bin_header(RCore *r, int mode) {
...
 }
 
+static int bin_hashes(RCore *r, int mode) {
+	ut64 lim = r_config_get_i (r->config, "cfg.hashlimit");
+	RIODesc *iod = r_io_desc_get (r->io, r->file->fd);
+	if (iod) {
+		r_bin_file_hash (r->bin, lim, iod->name);
+		const char *hashes = r_strbuf_get (r->bin->cur->o->info->hashes);
+		if (IS_MODE_JSON (mode)) {
+			PJ *pj = pj_new();
```
Comment (radare) [inline libr/core/cbin.c]: missing space and chk for null

**Comment by radare** [comment]
-t is not used but i dont see the realtionship with the 'hashing' concept. we can probably abuse `-hh`  but that looks ugly to me too. the -t thing sounds more to me like it should be used to show timestamps.. but i hate abusing flags that are already in there.. even more if its used to  show help

**Comment by devnexen** [comment]
In fact was running of single letters :-), you re right `t` does not relate to hashing.

**Comment by radare** [comment]
tired of waiting, ive been asking for 3 days to get a single proposal for an alternative here. merging :) thanks!

**Pull request merged**